sympy/core/basic.py

```diff
@@ -999,32 +1000,40 @@ def fallback(self, old, new):
             Try to replace old with new in any of self's arguments.
             """
             hit = False
-            args = list(self.args)
-            for i, arg in enumerate(args):
-                if not hasattr(arg, '_eval_subs'):
-                    continue
-                arg = arg._subs(old, new, **hints)
-                if not _aresame(arg, args[i]):
-                    hit = True
-                    args[i] = arg
-            if hit:
-                rv = self.func(*args)
-                hack2 = hints.get('hack2', False)
-                if hack2 and self.is_Mul and not rv.is_Mul:  # 2-arg hack
-                    coeff = S.One
-                    nonnumber = []
-                    for i in args:
-                        if i.is_Number:
-                            coeff *= i
+            def subs_old_new(self,old,new,hit):
+                for i, arg in enumerate(args):
```

Comment:
args is not a parameter of this method, and from the way you use it, it should be.